File: benchmarkDS/CoT/GEMINI_gemini-2.5-pro/Lse39.java
public class Lse39 {

    /**
     * Checks if the provided username and password are correct.
     * In a real-world application, this method would check credentials against a database
     * and use hashed passwords, not plaintext.
     *
     * @param username The username to check.
     * @param password The password to check.
     * @return A string indicating success or failure.
     */
    public static String login(String username, String password) {
        // NOTE: Hardcoding credentials is insecure and only for demonstration.
        final String CORRECT_USERNAME = "admin";
        final String CORRECT_PASSWORD = "password123";

        // Use .equals() for safe string comparison in Java.
        // A real application should use a constant-time comparison to prevent timing attacks.
        if (CORRECT_USERNAME.equals(username) && CORRECT_PASSWORD.equals(password)) {
            return "Login successful! Redirecting to the session page.";
        } else {
            return "Incorrect username or password.";
        }
    }

    public static void main(String[] args) {
        System.out.println("--- Login System Test Cases ---");

        // Test Case 1: Correct credentials
        System.out.println("\nTest Case 1 (Correct):");
        System.out.println("Input: user='admin', pass='password123'");
        System.out.println("Output: " + login("admin", "password123"));

        // Test Case 2: Incorrect password
        System.out.println("\nTest Case 2 (Incorrect Password):");
        System.out.println("Input: user='admin', pass='wrongpassword'");
        System.out.println("Output: " + login("admin", "wrongpassword"));

        // Test Case 3: Incorrect username
        System.out.println("\nTest Case 3 (Incorrect Username):");
        System.out.println("Input: user='user', pass='password123'");
        System.out.println("Output: " + login("user", "password123"));

        // Test Case 4: Both incorrect
        System.out.println("\nTest Case 4 (Both Incorrect):");
        System.out.println("Input: user='user', pass='wrongpassword'");
        System.out.println("Output: " + login("user", "wrongpassword"));

        // Test Case 5: Case-sensitive check
        System.out.println("\nTest Case 5 (Case-Sensitive):");
        System.out.println("Input: user='Admin', pass='password123'");
        System.out.println("Output: " + login("Admin", "password123"));
        
        System.out.println("\n--- End of Test Cases ---");
    }
}

File: benchmarkDS/Vanilla/OPENAI_gpt-5-2025-08-07/Lse39.c
#include <stdio.h>
#include <string.h>

const char* login(const char* username, const char* password) {
    if (username == NULL || password == NULL) {
        return "Incorrect username or password";
    }
    if (strcmp(username, "admin") == 0 && strcmp(password, "password123") == 0) {
        return "Redirecting to session page";
    }
    return "Incorrect username or password";
}

int main() {
    const char* tests[5][2] = {
        {"admin", "password123"},
        {"admin", "wrong"},
        {"user", "password123"},
        {"", "password123"},
        {"Admin", "password123"}
    };

    for (int i = 0; i < 5; i++) {
        printf("Test %d: %s\n", i + 1, login(tests[i][0], tests[i][1]));
    }

    return 0;
}